### PR TITLE
Rewrite an Evergreen test case as a Block Box Test

### PR DESCRIPTION
**Pull request by hlwarriner**
This change sets up some common functionality that Evergreen tests need in order to run on the Black Box Test framework and can hopefully be a model for other tests we want to migrate from
starboard/evergreen/testing/.

This rewrite gets around the logs reading/writing problem, which isn't supported for Black Box Tests currently, by instead relying on the updater status from the web API. This is probably better anyway since the web APIs are a well-defined interface for Cobalt (vs. logs) but some test cases may not be able to get the information they need from web APIs alone.

b/278953372

### CODE REVIEW

File: cobalt/black_box_tests/black_box_tests.py [hunk 1]
```diff
@@ -122,15 +131,28 @@ def tearDownClass(cls):
...
-        url=url,
-        target_params=all_target_params)
-    return new_runner
+
+    if poll_until_wait_seconds:
+      return black_box_cobalt_runner.BlackBoxCobaltRunner(
+          launcher_params=_launcher_params,
+          url=url,
+          target_params=all_target_params,
+          poll_until_wait_seconds=poll_until_wait_seconds,
```
Comment (kaidokert) [inline cobalt/black_box_tests/black_box_tests.py:148]: This seems like a pass-through here for `poll_until_wait_seconds`, i'd recommend just leaving it in "kwargs". E.g. only `else` part of this block should be needed and you should be able to drop `poll_until_wait_seconds` named arg from the function here.

File: cobalt/black_box_tests/tests/evergreen_verify_qa_channel_update_test.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          url=url,
+          target_params=['--update_check_delay_seconds=300'],
+          loader_target='loader_app') as runner:
+        runner.WaitForJSTestsSetup()
+        self.assertTrue(runner.JSTestsSucceeded())
+
+      url = server.GetURL(
+          file_name='testdata/evergreen_test.html?channel=test&status=Update installed, pending restart'  # pylint: disable=line-too-long
+          .replace(' ', '%20'))
+      # 100 seconds provides enough time for the initial update delay, the prod
```
Comment (johnxwork) [inline cobalt/black_box_tests/tests/evergreen_verify_qa_channel_update_test.py:41]: How does does it usually take to init? Is the 100 seconds dependent on network download?

File: cobalt/black_box_tests/black_box_cobalt_runner.py [hunk 3]
```diff
@@ -48,16 +48,26 @@ def __init__(self,
                url,
                log_file=None,
                target_params=None,
-               success_message=None):
+               success_message=None,
+               poll_until_wait_seconds=POLL_UNTIL_WAIT_SECONDS,
```
Comment (kaidokert) [inline cobalt/black_box_tests/black_box_cobalt_runner.py:52]: 👍 parametrizing timeouts is always good tweak, helps testability. (No action needed)

File: cobalt/black_box_tests/black_box_tests.py [hunk 4]
```diff
@@ -167,6 +189,31 @@ def LoadTests(launcher_params):
   return test_suite
 
 
+def LoadEvergreenEndToEndTests(launcher_params):
+  launcher = abstract_launcher.LauncherFactory(  # pylint: disable=unused-variable
+      launcher_params.platform,
+      'cobalt',
```
Comment (kaidokert) [inline cobalt/black_box_tests/black_box_tests.py:195]: Small readability nit: this gets a bit buried here, maybe pull it out into `EVERGREEN_TEST_BINARY='cobalt'` or such.

File: cobalt/black_box_tests/black_box_tests.py [hunk 5]
```diff
@@ -228,10 +275,20 @@ def Run(self):
...
     if (f'{_launcher_params.platform}/{_launcher_params.config}'
         in _DISABLED_BLACKBOXTEST_CONFIGS):
-      logging.warning('Blackbox tests disabled for platform:%s config:%s',
-                      _launcher_params.platform, _launcher_params.config)
+      run_cobalt_tests = False
+      logging.warning(
+          'Cobalt blackbox tests disabled for platform:%s config:%s',
+          _launcher_params.platform, _launcher_params.config)
+
+    if (f'{_launcher_params.platform}/{_launcher_params.config}'
```
Comment (kaidokert) [inline cobalt/black_box_tests/black_box_tests.py:287]: Readability nit: this got duplicated, pull it into a separate var: `launching_config=f'{_launcher_params.platform}/{_launcher_params.config}'`

File: cobalt/black_box_tests/black_box_tests.py [hunk 6]
```diff
@@ -243,12 +300,24 @@ def Run(self):
...
+        return_code = not unittest.TextTestRunner(
+            verbosity=2, stream=sys.stdout).run(suite).wasSuccessful()
+        return return_code
       else:
-        suite = LoadTests(_launcher_params)
-      # Using verbosity=2 to log individual test function names and results.
-      return_code = not unittest.TextTestRunner(
-          verbosity=2, stream=sys.stdout).run(suite).wasSuccessful()
-      return return_code
+        cobalt_tests_return_code = 0
```
Comment (kaidokert) [inline cobalt/black_box_tests/black_box_tests.py:307]: Do Evergreen tests need to run with a Proxy for any reason ? I'm almost sure they don't - I'd recommend getting it out of this `with Proxy` block - it's notoriously finicky and it's really only needed for _only_ Web Platform tests, even for other Blackboxtests it only adds pain and unreliability and i'm planning to break those out here sometime soon.

**Review by kaidokert — APPROVED** [review]
Overall looks like a great start. I'd just strongly suggest avoiding putting the proxy in loop here altogether ( b/170593065 )

File: cobalt/black_box_tests/black_box_tests.py (hunk 4 above)
Comment (hlwarriner) [reply cobalt/black_box_tests/black_box_tests.py]: Done

File: cobalt/black_box_tests/black_box_tests.py (hunk 5 above)
Comment (hlwarriner) [reply cobalt/black_box_tests/black_box_tests.py]: Good call, done.

File: cobalt/black_box_tests/black_box_tests.py (hunk 6 above)
Comment (hlwarriner) [reply cobalt/black_box_tests/black_box_tests.py:297]: I tried pulling it out of the `with ProxyServer` block but got this error in the test:
```
[MainWebModule/951761:0519/090427.096454:ERROR:browser_module.cc(1213)] NetFetcher error on http://127.0.0.1:37913/testdata/evergreen_test.html?resetInstallations=true: net::ERR_PROXY_CONNECTION_FAILED, response code -1
```

Can we leave it as is for now and consider removing the Proxy use for Evergreen test cases when it's removed for other tests?

By the way I seem to remember seeing an architecture diagram at some point under `cobalt/black_box_tests` but can't find anything now. I think having something like that would make it easier to contribute to the framework.

File: cobalt/black_box_tests/tests/evergreen_verify_qa_channel_update_test.py (hunk 2 above)
Comment (hlwarriner) [reply cobalt/black_box_tests/tests/evergreen_verify_qa_channel_update_test.py:41]: Yeah, the time to update is dependent on the download time. I was thinking we could start with this and then increase it if needed.

**Comment by hlwarriner** [comment]
Just noting that I expect the new test to fail for SB16 until b/283144901 is fixed. We can either wait to merge this or I can add a condition to disable the test for now (similar to what we did with https://github.com/youtube/cobalt/pull/394).

File: cobalt/black_box_tests/black_box_tests.py (hunk 6 above)
Comment (kaidokert) [reply cobalt/black_box_tests/black_box_tests.py:298]: Sorry i thought i replied here earlier already. If you want to get the code in and running, yes go ahead as is. I'm just expecting you'll be seeing a lot of flakiness trying to run with it, so we'd need to follow up pretty urgently.

The reason why you are getting `ERR_PROXY_CONNECTION_FAILED` is that the command line launching with it is likely still passing the `--proxy` args to Cobalt itself when it runs. I would think that should be not too tricky to prevent.

And yes, there definitely was an ASCII art diagram somewhere -  possibly the one added in go/cobalt-cl/115749 ( there's some scribbled up in http://go/cobalt-testing-pip but that's probably not what you are looking for. It does summarize a few issues with the framework though ). 

tl;dr if this runs as is for you go ahead and submit, don't break your back making it too clean.

Comment (hlwarriner) [reply cobalt/black_box_tests/black_box_tests.py:302]: Ok thanks yeah, I think I'll go ahead and get this in using proxy for now.

And that's the diagram I was thinking of, thanks!

**Comment by kaidokert** [comment]
> Sound ok to merge this @kaidokert ?

Yep, i approved above already anyway - go ahead
